src/esp/physics/CollisionGroupHelper.h

```diff
@@ -0,0 +1,207 @@
+   * @param groupB The group which will be masked.
+   */
+  static void setGroupInteractsWith(CollisionGroup groupA,
+                                    CollisionGroup groupB,
+                                    bool interacts) {
+    CollisionGroups groupAMask = collisionGroupMasks.at(groupA);
+    groupAMask = interacts ? groupAMask | groupB : groupAMask & ~groupB;
+    collisionGroupMasks.at(groupA) = groupAMask;
+  };
+
+  /**
+   * @brief Set the mask for a group defining which other groups it will
+   * interactive with.
+   *
+   * Mask check logic is two ways: AandBCollide = (GroupA & MaskB) && (GroupB &
+   * MaskA) Note: Editing default engine group (Default, Static, Kinematic,
+   * FreeObject, Robot, Noncollidable, AllFilter) behavior is discouraged.
+   *
+   * @param group The group to modify.
+   * @param mask The mask to apply.
+   */
+  static void setMaskForGroup(CollisionGroup group, CollisionGroups mask) {
+    collisionGroupMasks.at(group) = mask;
+  }
+
+  /**
+   * @brief Get the collision group by its name. Must pass a valid name.
+   *
+   * @param groupName The collision group's configured name.
+   */
+  static CollisionGroup getGroup(const std::string& groupName) {
+    ESP_CHECK(collisionGroupNames.count(groupName) != 0,
+              "Invalid groupName provided. Matches no CollisionGroup.");
+    return collisionGroupNames.at(groupName);
+  }
+
+  /**
+   * @brief Get a collision group's configured name.
+   *
+   * @param group The collision group.
+   * @return The group's configured name.
+   */
+  static std::string getGroupName(CollisionGroup group) {
+    for (std::map<std::string, CollisionGroup>::iterator it =
+             collisionGroupNames.begin();
+         it != collisionGroupNames.end(); ++it) {
+      if (group == it->second) {
+        return it->first;
+      }
+    }
+    // enum input, so should not get here unless the map is corrupted
+    CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+    return "";
+  }
+
+  /**
+   * @brief Set a custom name for a collision group.
+   *
+   * Fails if the desired name is already in use.
+   *
+   * @param group The collision group.
+   * @return Whether or not the set was successful.
+   */
+  static bool setGroupName(CollisionGroup group, const std::string& newName) {
+    auto currentName = getGroupName(group);
+    if (collisionGroupNames.count(newName) != 0) {
+      LOG(WARNING) << "CollisionGroupHelper::setGroupName - requested group "
+                      "name is already in use, aborting.";
+      return false;
+    }
+    collisionGroupNames[newName] = collisionGroupNames.at(currentName);
+    collisionGroupNames.erase(currentName);
+    return true;
+  }
+
+  /**
+   * @brief Get a list of all configured collision group names for easy
+   * iteration over groups.
+   *
+   * @return List of configured group names
+   */
+  static std::vector<std::string> getAllGroupNames() {
+    std::vector<std::string> groupNames(collisionGroupNames.size());
```

Comment:
wait, doesn't this create a vector that has double the items and the first half is empty strings?

(I think you meant to either call `reserve()` instead, or use `[]` instead of `push_back()`?)